### ISSUE TITLE
Metrics Stop Counting

### ISSUE THREAD

**Issue by seang96** [Cottand/leng#50]
I am using netdata to monitor metrics and the Served DNS Replies and Upstream DNS Requests after ~2.5-3 hours just stop recording. Restarting leng fixes it for a couple hours until it fails again. The go and process related metrics do continue to work; it seems to only effect the custom metrics.

I have an output of metrics endpoint and logs in a [gist here](https://gist.github.com/seang96/c627ba88acccdc23e5ccf9062666da6f).

**Comment by Cottand** [comment]
Hey @seang96 thanks for reporting this! I run leng with templating (so it gets restarted quite often) so it is possible I would not catch something like this.

When you say it stops counting, do you mean that the cardinality does not increase for those two metrics only? Or do existing metrics keep getting updated?

For example, is `leng_request_custom` updated over time, even after 3h? It is built into leng by me but unlike the others it is low cardinality.


This is what it looks like on the HTTP API:
```
HELP leng_request_custom Served custom DNS requests
# TYPE leng_request_custom counter
leng_request_custom 2
```

**Comment by seang96** [comment]
I currently don't have any custom DNS records configured so that metric isn't used, but the upstream DNS and the DNS requests I have been tracking and experienced this.

**Comment by Cottand** [comment]
Ok no worries. Can you confirm the metric `upstream_request_doh` works (for a while like you said) but then afterwards - even though upstream DoH requests keep being made - the metric count just stops increasing?

**Comment by Cottand** [comment]
I have put out a potential fix at https://github.com/Cottand/leng/pull/51.
It disables high cardinality metrics by default - you need to enable them again with
```toml
[Metrics]
    highCardinalityEnabled = true
```

Please run that for a while and let me know if your dashboards make sense.

### How to run leng from the PR:

- For Nix, use the Flake `github:cottand/leng/d596c52`

- For Docker, use `ghcr.io/cottand/leng:sha-d596c52`

**Comment by seang96** [comment]
Nice update and information! I'll test this out tonight.

As a side note does disabling high cardinality still record these metrics but without the labels for each domain, or does this disable the metrics entirely?

**Comment by Cottand** [comment]
The former! It leaves the `q_name` and the `remote_ip` blank but preserves the rest of the dimenions :+1:

**Comment by seang96** [comment]
This issue appears to be resolved for my setup with that PR!

**Comment by Cottand** [comment]
Amazing, thanks for reporting back!

**Comment by Cottand** [comment]
FYI I have now released v1.5.1 so you do not need to use d596c52 anymore
